m3u8/reader.go

```diff
@@ -616,6 +602,52 @@ func parseDateRange(line string) (*DateRange, error) {
 	return &dr, nil
 }
 
+func parseExtXStartParams(parameters string) (float64, bool, error) {
+	var startTime float64
+	var startTimePrecise bool
+	var err error
+
+	for _, attr := range decodeAttributes(parameters) {
+		switch attr.Key {
+		case "TIME-OFFSET":
```

Comment:
When parsing the time offset, should we return an error if the offset value does not conform to the spec?
from the RFC:
> The absolute value of TIME-OFFSET SHOULD NOT be larger than the
      Playlist duration.  If the absolute value of TIME-OFFSET exceeds
      the duration of the Playlist, it indicates either the end of the
      Playlist (if positive) or the beginning of the Playlist (if
      negative).
 > If the Playlist does not contain the EXT-X-ENDLIST tag, the TIME-
      OFFSET SHOULD NOT be within three Target Durations of the end of
      the Playlist file.

This might be overkill and hard to do with the structure of the packager, but could also be a nice to have feature.

Reply:
I think it would be useful, but, in my view, we should parse everything that is syntactically valid including if the version is wrong or some values too big or small, and then let the application, or a special validation phase handle such things.

Let's ponder it for later.

Reply:
That's a valid decision to make. In that case, feel free to merge whenever :)